oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -72,6 +73,10 @@ struct NodeInfo {
     /// (with one small exception, when the Runtime is in the process of closing
     /// down and the [`JoinHandle`] is held by the shutdown processing code).
     join_handle: Option<JoinHandle<()>>,
+
+    /// If the Node is currently running, holds the [`oneshot::Sender`] for
+    /// shutting down the async routine executed by the Nodes thread.
```

Comment:
This comment could be clearer I think; basically this is a notification object whose receiver is sent to the node, and this is the sender; the agreement is that the runtime will notify it upon termination and then start waiting on the join handle; it's up to the node to figure out how to actually terminate when receiving a notification.